src/connection-pool-party.js

```diff
@@ -474,6 +500,12 @@ export default class ConnectionPoolParty extends EventEmitter {
     return this.connectionPoolFactory(updatedDsn)
       .then(
         (pool) => {
+          if (pool.error) {
+            // some connection pool factories may opt to return an error instead
+            // of rejecting the promise. the existence of an error indicates that
+            // the pool did not heal successfully
+            return pool.error;
```

Comment:
shouldn't you throw here?

Reply:
nope, returning is intended. `_healPool` should never reject, only resolve with errors, which is what `_healPools` is expecting